### PR TITLE
Automate editorial style checking

### PR DESCRIPTION
**Pull request by juliamrch**
## Checklist

- [ ] My PR is related to an opened issue : Nope

**⚠️ If changes affect GitHub Action files**

- [x] I have added tests to cover my changes : Additional GitHub action on this repository [Checks on a file on a test repo](https://github.com/juliamrch/ci-cd-chaos/pull/45/files)

## Pull request type

Please check the type of change your PR introduces:

- [ ] Documentation content changes
- [ ] Bugfix on the site
- [ ] Build related changes
- [x] Other (please describe): Automate editorial and spelling review with [Vale](https://vale.sh)
      
## Description

_A summary of the changes_

This PR includes an additional GitHub Action with [Vale integration](https://github.com/errata-ai/vale-action). Vale is a documentation linter for editorial style and spelling to automate reviewing and ensure high quality content is merged.

It's configured as follows:

`.vale.ini` file to specify how to check the content:

```
StylesPath = .github/styles

Vocab = Doc

MinAlertLevel = suggestion

Packages = Hugo, Google

[*.md]
BasedOnStyles = Vale, Google
Google.Headings = NO

[*.xml]
Transform = docbook-xsl-snapshot/html/docbook.xsl
```

In the current file, Vale is configured as follows:

- Styles are placed in `/.github/` to allow running Vale on local and on GitHub Actions runners as well.
- Vocabulary list contains Clever Cloud specific terms that should be enforced or ignored (ex: `add-on` is accepted while `addon` isn't).
- Good style is based on [Vale built-in style rules](https://vale.sh/docs/topics/styles/#built-in-style) and [Google documentation style guide](https://developers.google.com/style).
- Style reference for headings as specified by Google is ignored (because it flags emojis in titles, which we use)
- Support for `.xml` has been included, because our files contains it.

The workflow will run on Pull Requests and fail if the submitted content contains 1 error, then post the error directly on the review, like in this example:

Capture d’écran 2024-02-07 à 11 59 59

The review scope is filtered by [Reviewdog](https://github.com/reviewdog/reviewdog?tab=readme-ov-file#filter-mode) when running as a GitHub Action. It's been set to `added` in this PR, to consider only changes in the modified or added lines.

### Further Customization

Vale can be tweaked and customized as we wish. We can add exceptions, specific vocabulary and rules that will suit our doc. This PR contains basic configuration that can be expanded.

### Drawbacks

There are some limitations in Vale: some lines happen to be missed and the[ reviewdog action fails if Vale flags something as a warning](https://github.com/errata-ai/vale-action/issues/84) (it should only fail on error). It's still in development and we could contribute to enhance it as well.

### Why is this needed?

Writing good doc is hard but makes the reading easy. Memorizing all good practice principles and reviewing/enforcing them manually takes time and isn't an option if we want to allow everyone to contribute. Including this changes will allow us to:

- target specific problems and enforce accessibility and clarity only on submitted content from contributors, who won't need to read the whole guides to make quick changes or new pages
- give feedback on PRs immediately so the authors can start fixing issues as soon as they submit it

## Reviewes
_Who should review these changes?_ @cnivolle @davlgd

### CODE REVIEW

File: .github/styles/Google/WordList.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  above: preceding
+  account name: username
+  action bar: app bar
+  admin: administrator
+  Ajax: AJAX
+  a\.k\.a|aka: or|also known as
+  Android device: Android-powered device
+  android: Android
+  API explorer: APIs Explorer
+  application: app
```
Comment (cnivolle) [inline .github/styles/Google/WordList.yml:32]: @juliamrch A question for the rules like the line 32: "app => application".
How can we override this kinds of rules in the future ?

**Review by cnivolle — DISMISSED** [review]
Tested and approved. Let's start with it now and refine the rules later. The suggestions given by Vale are helpful.

File: .github/styles/Google/WordList.yml (hunk 1 above)
Comment (davlgd) [reply .github/styles/Google/WordList.yml:32]: You can add `application` to the exception list (`.github/styles/config/vocabularies/Doc/accept.txt`), I tried it and it work after a `vale sync` and a reload of the file.

**Comment by davlgd** [comment]
LGTM, it's a good starting point. Maybe we can add a short mention to contributing rules and links to Vale, setup guide, list of integrations, etc. before merge.

After some tests it's good at spelling, but some rules are more annoying than useful and we'll need to choose what to keep or not in the coming weeks. But as they only appear as warnings, it's ok for now. For example, the future form or `our` are always marked as mistakes to avoid with little informations on why, but when you look at the detailed rule on an external website, it tells... "it's ok if you talk about future/your organisation" 😄 

So in the meantime, we should keep it clear that it's a first step with general rules we'll refine over time, and that contributor can choose to follow or not.

**Review by cnivolle — APPROVED** [review]

**Review by davlgd — APPROVED** [review]

**Pull request merged**